execute/factory.go

```diff
@@ -56,7 +57,7 @@ type PluginFactory struct {
 	homeChainReader   reader.HomeChain
 	estimateProvider  gas.EstimateProvider
 	tokenDataObserver tokendata.TokenDataObserver
-	contractReaders   map[cciptypes.ChainSelector]types.ContractReader
+	contractReaders   map[cciptypes.ChainSelector]contractreader.Reader
```

Comment:
I believe this would require coordinated changes in `chainlink` where this function is used

Reply:
Yeah preferably coordination not needed IMO

Reply:
Can `chainlink` not pass `map[cciptypes.ChainSelector]types.ContractReader` to `NewPluginFactory`? Not sure if this works or not.

Reply:
I'm pretty sure map values work a little different and you have to convert it manually
```
readers := make(map[cciptypes.ChainSelector]reader.ContractReaderFacade)
	for chain, cr := range p.contractReaders {
		readers[chain] = cr
	}
```

You can see the error by leaving the `types.ContractReader` type on `func NewPluginFactory(...)`

Reply:
This doesn't show any errors

```
type PluginFactory struct {
    contractReaders map[cciptypes.ChainSelector]contractreader.Reader
    // ... other stuff
}

func NewPluginFactory(
	lggr logger.Logger,
	ocrConfig reader.OCR3ConfigWithMeta,
	commitCodec cciptypes.CommitPluginCodec,
	msgHasher cciptypes.MessageHasher,
	homeChainReader reader.HomeChain,
	contractReaders map[cciptypes.ChainSelector]types.ContractReader,
	chainWriters map[cciptypes.ChainSelector]types.ChainWriter,
) *PluginFactory {
	readers := make(map[cciptypes.ChainSelector]contractreader.Reader)
	for chain, cr := range contractReaders {
		readers[chain] = cr
	}

	return &PluginFactory{
		lggr:            lggr,
		ocrConfig:       ocrConfig,
		commitCodec:     commitCodec,
		msgHasher:       msgHasher,
		homeChainReader: homeChainReader,
		contractReaders: readers,
		chainWriters:    chainWriters,
	}
}
```

Reply:
Right, you needed to map the types manually. The original changes would have required that mapping in `chainlink` code